dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/collection/DefaultCollectionService.java

```diff
@@ -178,75 +162,71 @@ public void delCollectionItems( IdentifiableObject object, String propertyName,
 
         if ( itemCodes.isEmpty() )
         {
-            return;
+            return TypeReport.empty( property.getItemKlass() );
         }
 
+        TypeReport report = new TypeReport( property.getItemKlass() );
         manager.refresh( object );
 
         if ( property.isOwner() )
         {
-            delOwnedCollectionItems( object, property, itemCodes );
+            delOwnedCollectionItems( object, property, itemCodes, report );
         }
         else
         {
-            delNonOwnedCollectionItems( object, property, itemCodes );
+            delNonOwnedCollectionItems( object, property, itemCodes, report );
         }
 
         validateAndThrowErrors( () -> schemaValidator.validateProperty( property, object ) );
         manager.update( object );
 
         dbmsManager.clearSession();
         cacheManager.clearCache();
+        return report;
     }
 
-    private void delOwnedCollectionItems( IdentifiableObject object, Property property, Collection<String> itemCodes )
+    private void delOwnedCollectionItems( IdentifiableObject object,
+        Property property,
+        Collection<String> itemCodes,
+        TypeReport report )
         throws Exception
     {
         Collection<IdentifiableObject> collection = getCollection( object, property );
 
-        for ( IdentifiableObject item : getItems( property, itemCodes ) )
-        {
-            collection.remove( item );
-        }
+        updateCollectionItems( property, itemCodes, report, ErrorCode.E1109, collection::remove );
     }
 
     private void delNonOwnedCollectionItems( IdentifiableObject object, Property property,
-        Collection<String> itemCodes )
+        Collection<String> itemCodes, TypeReport report )
     {
         Schema owningSchema = schemaService.getDynamicSchema( property.getItemKlass() );
         Property owningProperty = owningSchema.propertyByRole( property.getOwningRole() );
 
-        for ( IdentifiableObject item : getItems( property, itemCodes ) )
-        {
-            try
-            {
+        updateCollectionItems( property, itemCodes, report, ErrorCode.E1109,
+            item -> {
                 Collection<IdentifiableObject> collection = getCollection( item, owningProperty );
 
                 if ( collection.contains( object ) )
                 {
                     collection.remove( object );
-                    validateAndThrowErrors( () -> schemaValidator.validateProperty( owningProperty, item ) );
+                    validateAndThrowErrors( () -> schemaValidator.validateProperty( property, item ) );
```

Comment:
Just want to highlight the change here from `owningProperty` to `property`. 
This wasn't the issue originally identified but if the code in _add_ needs this change so should the same code for _delete_